src/Cake.Core/Polyfill/EnvironmentHelper.cs

```diff
@@ -119,7 +119,7 @@ public static Runtime GetRuntime()
         public static FrameworkName GetBuiltFramework()
         {
 #if NETCORE
-            return new FrameworkName(".NETStandard,Version=v2.0");
+            return new FrameworkName(".NETCoreApp,Version=v2.1");
```

Comment:
So
* Cake.CoreCLR = netcoreapp2.0
* Cake.Tool = netcoreapp2.1
So why not just a have an define on Cake.Tool and if def, no need for reflection?